src/main/java/org/fenixedu/academic/domain/accounting/Event.java

```diff
@@ -1147,52 +1150,39 @@ public boolean isOpenAndAfterDueDate(DateTime when) {
                 .isAfter(d.getDueDate()));
     }
 
+    @Deprecated
     @Atomic(mode = Atomic.TxMode.WRITE)
     public Refund refund(User creator, EventExemptionJustificationType justificationType, String reason) {
+        final DateTime now = new DateTime().minusSeconds(2);
+        DebtInterestCalculator debtInterestCalculator = getDebtInterestCalculator(now);
+        return refund(creator, justificationType, reason, debtInterestCalculator.getPaidDebtAmount());
+    }
+
+    @Atomic(mode = Atomic.TxMode.WRITE)
+    public Refund refund(User creator, EventExemptionJustificationType justificationType, String reason, BigDecimal value) {
         if (!isRefundable()) {
             throw new DomainException("error.event.cannot.be.refunded");
         }
+        if (value.signum() == 0) {
+            throw new DomainException("error.accounting.refund.not.possible.zero.value");
+        }
 
         //force open state before exemption creation
         super.setEventState(EventState.OPEN);
         
-        final DateTime now = new DateTime().minusSeconds(2);
+        final DateTime now = new DateTime().minusSeconds(3);
         
         DebtInterestCalculator debtInterestCalculator = getDebtInterestCalculator(now);
-        debtInterestCalculator.getRefunds().findAny().ifPresent(refund -> {
-            throw new DomainException(Optional.of(Bundle.ACCOUNTING), "error.accounting.refund.not.possible", refund.getCreated
-                    ().toString("dd-MM-yyyy HH:mm:ss"));
-        });
-        final BigDecimal paidDebtAmount = debtInterestCalculator.getPaidDebtAmount();
-        final Refund refund = new Refund(this, new Money(paidDebtAmount), creator, false, now);
-        
-        debtInterestCalculator = getDebtInterestCalculator(now);
-
-        final BigDecimal dueAmount = debtInterestCalculator.getDueAmount();
-        final BigDecimal dueFineAmount = debtInterestCalculator.getDueFineAmount();
-        final BigDecimal dueInterestAmount = debtInterestCalculator.getDueInterestAmount();
-        boolean createdInterestOrFineExemptions = false;
-
-        if (dueInterestAmount.signum() == 1) {
-            FixedAmountInterestExemption fixedAmountInterestExemption =
-                    new FixedAmountInterestExemption(this, creator.getPerson(), new Money(dueInterestAmount), justificationType,
-                            now, reason);
-            fixedAmountInterestExemption.setWhenCreated(now.plusSeconds(1));
-            createdInterestOrFineExemptions = true;
+        final BigDecimal paidDebtAmount = debtInterestCalculator.getPaidDebtAmount().subtract(debtInterestCalculator.getTotalRefundAmount());
```

Comment:
this is because several refunds can be made, and as such we should subtract the amount already refunded from the amount payed